unblob/file_utils.py

```diff
@@ -368,12 +371,33 @@ def chop_root(path: Path):
     return Path("/".join(relative_parts))
 
 
+def make_lost_and_found_path(path: Path) -> Path:
+    """Make a human readable, safe path."""
+    dir_path = path.parent
+
+    # . and .. would not be a valid filename, but they would lead to confusion
+    filename = {".": "dot", "..": "dot-dot"}.get(path.name, path.name)
+    dir_hash = hashlib.sha224(str(dir_path).encode(errors="ignore")).hexdigest()
+
+    # adapted from https://stackoverflow.com/questions/5574042/string-slugification-in-python
+    dir_slug = str(dir_path)
+    dir_slug = unicodedata.normalize("NFKD", dir_slug)
+    dir_slug = dir_slug.encode("ascii", "ignore").lower()
+    dir_slug = re.sub(rb"[^a-z0-9]+", b"-", dir_slug).strip(b"-")
+    dir_slug = re.sub(rb"[-]+", b"-", dir_slug).decode()
+
+    return Path(f".unblob-lost+found/{dir_slug}_{dir_hash}/{filename}")
```

Comment:
I would swap dir_slug and dir_hash, as that would be easier to read the slug/filename assuming most slug would be the original directory:
```
UNBLOB_LOST_FOUND = ".unblob-lost+found"

...


return Path(UNBLOB_LOST_FOUND) / f"{dir_hash}_{dir_slug}" / filename
```